internal/engine/upper.go

```diff
@@ -309,6 +308,24 @@ func handleBuildCompleted(ctx context.Context, engineState *store.EngineState, c
 		}
 	}
 
+	// Track the container ids that have been live-updated whether the
+	// build succeeds or fails.
```

Comment:
...why would we track a live-updated CID if the build failed?

Reply:
This is the distinction we spent a long time talking about on the other PR. If a user Run step fails on a syntax error, that is normal and expected. The container files have still diverged

Reply:
oh i see! well shoot, [we don't return a result in that case :/ ](https://github.com/windmilleng/tilt/blob/96e48fcb02bced80fe58c0e975f1d52655fe49e9/internal/engine/live_update_build_and_deployer.go#L205) -- can you make a ticket for this?

Reply:
filed as https://app.clubhouse.io/windmill/story/3094/live-update-should-still-work-if-run-steps-fail